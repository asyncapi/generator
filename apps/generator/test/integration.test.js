/**
 * @jest-environment node
 */

const { readFile } = require('fs').promises;
const path = require('path');
const Generator = require('../lib/generator');
const dummySpecPath = path.resolve(__dirname, './docs/dummy.yml');
const refSpecPath = path.resolve(__dirname, './docs/apiwithref.json');
const refSpecFolder = path.resolve(__dirname, './docs/');
const crypto = require('crypto');
const mainTestResultPath = 'test/temp/integrationTestResult';
//we do not want to download chromium for html-template if it is not needed
process.env['PUPPETEER_SKIP_CHROMIUM_DOWNLOAD'] = true;

describe('Integration testing generateFromFile() to make sure the result of the generation is not changend comparing to snapshot', () => {
  const generateFolderName = () => {
    //you always want to generate to new directory to make sure test runs in clear environment
    return path.resolve(mainTestResultPath, crypto.randomBytes(4).toString('hex'));
  };

  jest.setTimeout(60000);

  it('generated using Nunjucks template', async () => {
    const outputDir = generateFolderName();
    const generator = new Generator('@asyncapi/html-template@0.18.0', outputDir, { forceWrite: true, templateParams: { singleFile: true } });
    await generator.generateFromFile(dummySpecPath);
    const file = await readFile(path.join(outputDir, 'index.html'), 'utf8');
    expect(file).toMatchSnapshot();
  });

  it('generate using React template', async () => {
    const outputDir = generateFolderName();
    const generator = new Generator('@asyncapi/markdown-template@0.13.0', outputDir, { forceWrite: true });
    await generator.generateFromFile(dummySpecPath);
    const file = await readFile(path.join(outputDir, 'asyncapi.md'), 'utf8');
    expect(file).toMatchSnapshot();
  });

  it('generate json based api with referenced JSON Schema', async () => {
    const outputDir = generateFolderName();
    const generator = new Generator('@asyncapi/html-template@0.18.0', outputDir, {
      mapBaseUrlToFolder: { url: 'https://schema.example.com/crm/', folder: `${refSpecFolder}/`},
      forceWrite: true,
      templateParams: { singleFile: true }
    });
    await generator.generateFromFile(refSpecPath);
    const file = await readFile(path.join(outputDir, 'index.html'), 'utf8');
    expect(file).toMatchSnapshot();
  });
});
