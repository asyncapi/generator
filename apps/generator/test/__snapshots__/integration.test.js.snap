// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration testing generateFromFile() to make sure the result of the generation is not changend comparing to snapshot generate json based api with referenced JSON Schema 1`] = `
"This is a markdown file for my application.
App name is: **This is Async API with a schema reference**
Version v1 running on production mode 
"
`;

exports[`Integration testing generateFromFile() to make sure the result of the generation is not changend comparing to snapshot generate using React template 1`] = `
"This is a markdown file for my application.
App name is: **Dummy example with all spec features included**
Version v1 running on production mode 
"
`;

exports[`Integration testing generateFromFile() to make sure the result of the generation is not changend comparing to snapshot generated using Nunjucks template 1`] = `
"This is a markdown file for my application.
App name is: **Dummy example with all spec features included**
Version v1 running on production mode"
`;
